iree/hal/vulkan/renderdoc_capture_manager.cc

```diff
@@ -44,21 +54,11 @@ Status RenderDocCaptureManager::Connect() {
     return OkStatus();
   }
 
-// Load RenderDoc's library object.
-#if defined(IREE_PLATFORM_WINDOWS)
-  // TODO(scotttodd): Windows, ::LoadLibraryA()
-  return UnimplementedErrorBuilder(IREE_LOC);
-#else
-  renderdoc_library_ = ::dlopen("librenderdoc.so", RTLD_NOW);
-  if (renderdoc_library_ == nullptr) {
-    return NotFoundErrorBuilder(IREE_LOC)
-           << "Could not load librenderdoc.so. Is it on LD_LIBRARY_PATH?";
-  }
-#endif
+  ASSIGN_OR_RETURN(renderdoc_library_,
+                   DynamicLibrary::Load(absl::MakeSpan(kRenderDocSearchNames)));
 
-  // Fetch the API object from the loaded library.
   pRENDERDOC_GetAPI RENDERDOC_GetAPI =
-      (pRENDERDOC_GetAPI)::dlsym(renderdoc_library_, "RENDERDOC_GetAPI");
+      (pRENDERDOC_GetAPI)renderdoc_library_->GetSymbol("RENDERDOC_GetAPI");
```

Comment:
if you add a GetSymbol<T> override to DynamicLibrary then you can do this without the C-style case (GetSymbol<pRENDERDOC_GetAPI>("RENDERDOC_GetAPI");

Reply:
Pushed some changes. How's this look?